nanoFramework.WebServer/WebServer.cs

```diff
@@ -665,6 +641,43 @@ protected virtual void InvokeRoute(CallbackRoutes route, HttpListenerContext con
             route.Callback.Invoke(null, new object[] { new WebServerEventArgs(context) });
         }
 
+        /// <summary>
+        /// Logs a message.
+        /// </summary>
+        /// <param name="message">The message to be logged.</param>
+        protected virtual void Log(string message)
+        {
+            Debug.WriteLine(message);
+        }
+
+        private static void HandleContextResponse(HttpListenerContext context)
+        {
+            // When context has been handed over to WebsocketServer, it will be null at this point
+            // Do nothing this is a websocket that is managed by a websocketserver that is responsible for the context now. 
+            if (context.Response == null)
+            {
+                return;
+            }
+
+            try
+            {
+                context.Response.Close();
+            }
+            catch
+            {
+                // Nothing on purpose
```

Comment:
Note for another PR: we should add an event or equivalent here in this function all up. Because it may be due to a wifi disconnection or equivalent and the device will continue to run but the network won't and the webserver will stop.

Reply:
Few more from me:
Maybe we should introduce, few more default behaviours for WebServer:
- When there is no route, we could return 404
- When there is error in method invoked - return 500

Reply:
Great list! Let's go for this PR first :-) And let's see how to continue to improve it ;-)

Reply:
And one more:
- If we are using controllers do we need "CommandReceived" event? Maybe we should decouple logic and create two separate web servers - one with controllers and one with "CommandReceived". Or maybe just get rid of CommandReceived? I'm not sure if full .net supports such case as "CommandReceived"?

Reply:
We need that one with the controllers. I've been using this one to trap favicon, style sheets and other elements. So this is a necessary one, regardless of using controllers or not.

Reply:
Ahh, Ok.